### ISSUE TITLE
Implement @obsolete TypeScript Decorator

### ISSUE THREAD

**Issue by eonarheim** [excaliburjs/Excalibur#684]
#### Context

Currently it is difficult to communicate when methods or features in Excalibur are obsolete in a consistent way. The generated documentation is the only consistent way to know if a method is deprecated. This may catch people off guard when trying to use Excalibur over time without checking the docs. 

#### Proposal

We should use the new TypeScript/ES6 feature to implement this logging and also information. Possibly extract the method name and use the ex.Logger to log to the console or other appenders.

Something like this

```typescript
class Actor {
   @obsolete("Please use someNewMethod()")
   public someOldMethod(){
      ...
   }
}

```

```console
[warn] Deprecated method `someOldMethod` : Please use someNewMethod()
```

See information about decorators here:
https://www.typescriptlang.org/docs/handbook/decorators.html

**Comment by hogart** [comment]
There's whole library of useful decorators: https://github.com/jayphelps/core-decorators.js. While dragging entire library can be overkill, looking at their implementation can be useful, since they obviously fixed a lot of egde cases.

**Comment by eonarheim** [comment]
I've got something working, turned out to be slightly more difficult than I imagined but it is pretty cool.

@hogart Thanks! I used that library to get started, I had to make some modifications where getters and setters were concerned. Also couldn't use the object spread operator just yet (we are tied to the version of typescript that the lastest version of typedoc implements sadly)

@excaliburjs/core-contributors Let me know what you think
![image](https://cloud.githubusercontent.com/assets/612071/21076198/1437d48e-beeb-11e6-8d6d-dee1e47ecad7.png)

![image](https://cloud.githubusercontent.com/assets/612071/21076201/2609c532-beeb-11e6-808e-c3eae00f518d.png)
